File: book/python_algorithm_interview/Q/13.py
from typing import Optional

# Definition for singly-linked list.


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        listed = []

        node = head

        while node != None:
            listed.append(node.val)
            node = node.next

        left, right = 0, len(listed) - 1
        while left < right:
            if listed[left] != listed[right]:
                return False
            left += 1
            right -= 1
        return True

File: book/python_algorithm_interview/Q/13.ts
// Definition for singly-linked list.

class ListNode {
  val: number;
  next?: null;
  constructor(val: number = 0, next = null) {
    this.val = val;
    this.next = next;
  }
}

class Solution {
  isPalindrome(head: ListNode): boolean {
    let listed = [];

    let node = head;

    while (node != null) {
      listed.push(node.val);
      node = node.next;
    }

    let left = 0;
    let right = listed.length - 1;
    while (left < right) {
      if (listed[left] != listed[right]) {
        return false;
      }
      left += 1;
      right -= 1;
    }

    return true;
  }
}
